dist/ng-image-input-with-preview.js

```diff
@@ -183,6 +184,44 @@
               });
               return deferred.promise;
             };
+            ngModel.$asyncValidators.size = function(modelValue, viewValue) {
+              if (!attrs.size) {
+                return createResolvedPromise();
+              }
+  
+              var value = modelValue || viewValue;
+  
+              if (!value || !value.fileReaderPromise) {
+                return createResolvedPromise();
+              }
+  
+              var deferred = $q.defer();
+  
+              value.fileReaderPromise.then(function(dataUrl) {
+  
+                var image = document.createElement('img');
+  
+                image.addEventListener('load', function() {
+                  var valid = $scope.sizeRestrictions({
+                    size: viewValue.size
+                  });
+  
+                  $scope.$apply(function() {
+                    if (valid) {
+                      deferred.resolve();
+                    } else {
+                      deferred.reject('Invalid size');
+                    }
+                  });
+  
+                });
+  
+                image.src = dataUrl;
+              }, function() {
+                deferred.reject('Failed to detect size');
+              });
+              return deferred.promise;
+            };
```

Comment:
Not blaming you as this is obviously mostly copy-paste from above, but the noise to signal ratio in this function is pretty bad.

Reply:
Indeed